### ISSUE TITLE
How to use insertGraph (or upsert) with HasOneRelation?

### ISSUE THREAD

**Issue by angela-mylo** [Vincit/objection.js#678]
I can't seem to figure out how to make ```insertGraph``` work with a ```HasOneRelation```.  I am looking for a single method that can handle both creating a new child, or relating an existing child.

**Models:**
```
class Address extends Model {
  static get tableName () {
    return 'address'
  }
}

class Person extends Model {
  static get tableName () {
    return 'person'
  }

  static get relationMappings () {
    return {
      address: {
        relation: Model.HasOneRelation,
        modelClass: require('./address'),
        join: {
          from: 'person.addressId',
          to: 'address.id'
        }
      }
    }
  }
}
```

**Relate an existing record:**
```
Person.query()
  .insertGraph({
    fullName: 'Person with existing address', 
    address: {id: 123}
  }, {relate: true})
```
No errors returned.  A new address record is inserted into the database (not desired), and the addressId on the person table remains null.

If I try the same data with the ```upsertGraph``` method, then I get the following error:
```Error: undefined passed as argument #1 for 'whereComposite' operation. Call skipUndefined() method to ignore the undefined values.```

**Create a new record:**
```
Person.query()
  .insertGraph({
    fullName: 'Person with new address', 
    address: {addressLine: '123 Sesame street'}
  }, {relate: true})
```
The new address is correctly created, but not related to the person.  Same thing happens with ```upsertGraph``` method, 

Can anyone help?

**Comment by koskimas** [comment]
This should be fixed now in objection 0.9.4. Relating `HasManyRelation` or `HasOneRelation` is not possible using `insertGraph` and should now throw an exception when you try to do that. `upsertGraph` should work though.

**Comment by koskimas** [comment]
And actually you need to use BelongsToOneRelation in your example (because the foreigh key is in the owner model). With that everything should work.
